### PR TITLE
refactor: Upgrade mobile token 2.1.4

### PR DESCRIPTION
**Pull request by gorandalum**
Closes https://github.com/AtB-AS/kundevendt/issues/16909

- Add handling of error resolutions in errors from sdk. As of now
only handling the reset resolution, which signals that the sdk is
stuck. The rest of the errors doesn't have any specific handling
for now and are mainly treated as before.
- The new SDK version has the possibility for both local logger
and remote logger. For now, the previously used logger is passed in
as local logger, and for remote logger we just send the errors
as notify events to Bugsnag. We will see how this turns out in
Bugsnag and if it needs to be revised.
- Added some more libs in the Podfile, which is required
dependencies for the SDK, even if we don't use them yet.

Depends on: https://github.com/AtB-AS/abt-token-server/pull/68

### CODE REVIEW

**Review by hanne-at-atb — APPROVED** [review]
Things are looking good to me, but probably nice to have another pair of eyes on this since I didn't work too much with the app lately 🌟

File: src/mobile-token/MobileTokenContext.tsx [hunk 1]
```diff
@@ -121,11 +122,12 @@ export const MobileTokenContextProvider: React.FC = ({children}) => {
     } else {
       // We've finished with remote tokens. Cancel timeout notification.
       cancelTimeoutHandler?.();
+      setIsTimout(false);
     }
   }, [enabled, nativeTokenStatus, token_timeout_in_seconds]);
 
   useInterval(
-    () => checkRenewMutate(nativeToken),
+    () => checkRenewMutate({token: nativeToken, traceId: 'uuid'}),
```
Comment (jorelosorio) [inline src/mobile-token/MobileTokenContext.tsx:130]: should 'uuid' be a variable instead?

File: src/mobile-token/hooks/use-load-native-token-query.tsx [hunk 2]
```diff
@@ -36,6 +41,7 @@ export const useLoadNativeTokenQuery = (
       }
     },
     enabled,
+    retry: 0,
```
Comment (jorelosorio) [inline src/mobile-token/hooks/use-load-native-token-query.tsx:44]: No retries?

**Comment by jorelosorio** [comment]
@gorandalum I am not sure if there are breaking changes by doing an update to:

```
New version of the mobile client SDK: 2.1.1
* Make device info optional for barcodes in the beginInspection function
  * Opting out of device info reduces the data size in the barcodes by around 40%
  * The recommended config for device info is to set the application and os fields to false
* Added iOS privacy manifest
```

Seems like there are some improvements there.

File: src/mobile-token/hooks/use-preemptive-renew-token-mutation.tsx [hunk 3]
```diff
@@ -43,5 +52,22 @@ export const usePreemptiveRenewTokenMutation = (userId?: string) => {
         );
       }
     },
+    onError: async (err: any, {token, traceId}) => {
+      notifyBugsnag(err, {
+        errorGroupHash: 'AuthError',
```
Comment (jorelosorio) [inline src/mobile-token/hooks/use-preemptive-renew-token-mutation.tsx:57]: @reidzeibel Could you please confirm this is the right group hash for mobile-token? :)

Comment (reidzeibel) [reply src/mobile-token/hooks/use-preemptive-renew-token-mutation.tsx:57]: We have these:

`TokenLoadingTimeoutError`
`GetSignedTokenError`

on this PR : https://github.com/AtB-AS/mittatb-app/pull/4439/files#diff-fd9df4f15f1da5e1656210163837e5dd7992c4790f04cd721596106c8813a7f4

We have no grouping yet for this one, but I think it shouldn't be `AuthError`

**Review by reidzeibel — CHANGES_REQUESTED** [review]

File: src/mobile-token/hooks/use-preemptive-renew-token-mutation.tsx (hunk 3 above)
Comment (jorelosorio) [reply src/mobile-token/hooks/use-preemptive-renew-token-mutation.tsx]: Fixed

**Review by reidzeibel — APPROVED** [review]
Looks good, in the future we might want to revisit this error group hash for mobile token.

**Pull request merged**